sc/src/test/java/uk/ac/ebi/atlas/metadata/CellMetadataDaoIT.java

```diff
@@ -0,0 +1,149 @@
+    public void validCellIdHasMetadataValues(String experimentAccession) {
+        String cellId = jdbcUtils.fetchRandomCellFromExperiment(experimentAccession);
+
+        LOGGER.info("Retrieving metadata field names for experiment " + experimentAccession);
+        List<SingleCellAnalyticsCollectionProxy.SingleCellAnalyticsSchemaField> metadataFieldNames = subject.getMetadataFieldNames(experimentAccession);
+
+        assertThat(metadataFieldNames).isNotEmpty();
+
+        metadataFieldNames.forEach(field -> {
+            LOGGER.info("Retrieving values for " + field.displayName() + " metadata for cell ID " + cellId + " from experiment " + experimentAccession);
+
+            assertThat(subject.getMetadataValueForCellId(experimentAccession, field, cellId)).isPresent();
+        });
+    }
+
+    @ParameterizedTest
+    @MethodSource("experimentsWithFactorsProvider")
+    public void validExperimentAccessionHasFactorFields(String experimentAccession) {
+        String cellId = jdbcUtils.fetchRandomCellFromExperiment(experimentAccession);
+
+        LOGGER.info("Retrieving factor fields for cell ID " + cellId + " from experiment " + experimentAccession);
+        assertThat(subject.getFactorFieldNames(experimentAccession, cellId)).isNotEmpty();
+    }
+
+    @Test
+    public void invalidCellIdAndExperimentAccessionHasNoFactorFields() {
+        String experimentAccession = RandomDataTestUtils.getRandomExperimentAccession();
+        String cellId = "FOO";
+
+        assertThat(subject.getFactorFieldNames(experimentAccession, cellId)).isEmpty();
+    }
+
+    @ParameterizedTest
+    @MethodSource("experimentsWithMetadataProvider")
+    public void validCellIdsHaveMetadataValues(String experimentAccession) {
+        List<String> cellIds = jdbcUtils.fetchRandomListOfCellsFromExperiment(experimentAccession, ThreadLocalRandom.current().nextInt(1, 2000));
+
+        LOGGER.info("Retrieving metadata field names for experiment " + experimentAccession);
+        List<SingleCellAnalyticsCollectionProxy.SingleCellAnalyticsSchemaField> metadataFieldNames = subject.getMetadataFieldNames(experimentAccession);
+
+        assertThat(metadataFieldNames).isNotEmpty();
+
+        metadataFieldNames.forEach(field -> {
+            LOGGER.info("Retrieving values for " + field.displayName() + " metadata for " + cellIds.size() + " random cell IDs from experiment " + experimentAccession);
+
+            assertThat(subject.getMetadataValueForCellIds(experimentAccession, field, cellIds))
+                    .isNotEmpty()
+                    .containsKeys(cellIds.toArray(new String[0]));
+        });
+
+    }
+
+    @Test
+    public void getFieldValuesForNoFieldsReturnsEmpty() {
+        String experimentAccession = jdbcUtils.fetchRandomSingleCellExperimentAccession();
+
+        assertThat(subject.getQueryResultForMultiValueFields(experimentAccession, Optional.empty())).isEmpty();
+    }
+
+    private Iterable<String> experimentsWithMetadataProvider() {
```

Comment:
In parameterized tests [method sources can return `Stream<T>`](https://junit.org/junit5/docs/current/user-guide/#writing-tests-parameterized-tests-sources-MethodSource), saving you the `collect(...)` step. If you want to return unique values use [`distinct()`](https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html#distinct--).